### DISCUSSION TITLE
feat(BootstrapInput): BootstrapInput Bind non-nullable TValue int, float, double, decimal Setting Default Value to Zero When Null

### DISCUSSION THREAD

**Discussion by mrtonyguo** [dotnetcore/BootstrapBlazor#3953] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

复现步骤：

1. 附件最小Demo，打开Home.razor页面， Demo里测试绑定的泛型数据为 不可为Null的decimal， 在BootstrapInput框里输入 数字 300，然后全选中数字，执行一次性删除，在下面div标签里能看到绑定的decimal值还是300；
![image](https://github.com/user-attachments/assets/3ee845d2-ad90-4400-98ba-f89afef55621)
![image](https://github.com/user-attachments/assets/6e78c601-0771-4033-8534-787fc8e04a39)

Steps:

1. Open the Home.razor page with the min demo in the attachment. The data bind-value for evaluation in the demo is a non-nullable decimal. 
2. Enter the number 300 in the BootstrapInput Components, then select all the numbers and perform a one-time deletion. 
3. You can see that the bound decimal value is still 300 in the div tag below;

### Describe the solution you'd like

1. 附件请查收Demo

2. 经查阅源代码，是否可以在ValidateBase.cs绑定的泛型CurrentValueAsString里新增对于不可为Null的数字类型进行判断，如果为空值则等于对应类型的默认值，添加如下代码，经测试可行，烦请审阅：
![image](https://github.com/user-attachments/assets/8c8ec0c4-2bce-4acd-a162-fed0b143d162)

`            else if (typeof(TValue).IsValueType && Nullable.GetUnderlyingType(typeof(TValue)) == null && string.IsNullOrEmpty(value))
            {
                // Non-nullable value type (e.g., int, double, etc.)
                PreviousParsingAttemptFailed = false;
                CurrentValue = default!;
            }`
[BootstrapBlazor.Net8.zip](https://github.com/user-attachments/files/16380844/BootstrapBlazor.Net8.zip)

**Comment by bb-auto** [comment]
@mrtonyguo Thank you for contacting us. We will give feedback later.

**Comment by ArgoZhang** [comment, marked as answer]
@mrtonyguo 抱歉回复的晚了

### 绑定类型为不可为空类型

点击组件获取焦点，全选数字，按退格键或者删除键，全部删除时，网页为 `null` 值，空值对不可为空类型是非法值，所以恢复 `500` 如果改成 `0` 这并不合理，可能对某些需求合理 

### 绑定类型为可为空类型

全部删除后网页为 `null` 值，数值也为 `null` 值无歧义

### 总结

歧义是网页返回 `null` 时，是保持原有值，还是变成绑定类型默认值

这里可以设置一个参数，两种需求都合理。

暂定参数为 `AutoSetDefaultWhenNull` 默认 false 保持目前行为，设置为 `true` 时符合你的预期

已发布 `v8.7.5`

**Reply by TonyGuo16** [reply]
感谢大佬，这样使用更加灵活了

**Reply by densen2014** [reply]
tony 是你吗？

**Reply by mrtonyguo** [reply]
> tony 是你吗？

是的，大佬
